File: VB-Examples/19_PivotTables/ClearPivotFields.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet.PivotTables

Namespace ClearPivotFields
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            'Create a new instance of Workbook
            Dim workbook As New Workbook()

            'Load the workbook from the specified file path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\PivotTableExample.xlsx")

            'Get the worksheet named "PivotTable" from the workbook
            Dim sheet As Worksheet = workbook.Worksheets("PivotTable")

            'Cast the first PivotTable in the worksheet to XlsPivotTable data type
            Dim pt As XlsPivotTable = TryCast(sheet.PivotTables(0), XlsPivotTable)

            'Clear all data fields in the PivotTable
            pt.DataFields.Clear()

            'Calculate the data in the PivotTable
            pt.CalculateData()

            'Specify the filename for the resulting workbook that will be saved
            Dim result As String = "ClearPivotFields_result.xlsx"

            'Save the workbook to the specified file path in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/19_PivotTables/ClearPivotFields.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet.PivotTables;

namespace ClearPivotFields
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load an excel file including pivot table
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\PivotTableExample.xlsx");

            // Get the sheet in which the pivot table is located
            Worksheet sheet = workbook.Worksheets["PivotTable"];

            // Get the first pivot table from the sheet
            XlsPivotTable pt = sheet.PivotTables[0] as XlsPivotTable;

            // Clear all the data fields in the pivot table
            pt.DataFields.Clear();

            // Calculate the pivot table data
            pt.CalculateData();

            // Specify the filename for the resulting workbook
            string result = "ClearPivotFields_result.xlsx";

            // Save the modified workbook to the file
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
